packages/attestation-service/src/models/attestation.ts

```diff
@@ -28,16 +46,46 @@ export default (sequelize: Sequelize) => {
     account: DataTypes.STRING,
     identifier: DataTypes.STRING,
     issuer: DataTypes.STRING,
+    countryCode: DataTypes.STRING,
+    phoneNumber: DataTypes.STRING,
+    message: DataTypes.STRING,
+    ongoingDeliveryId: DataTypes.STRING,
+    providers: DataTypes.STRING,
+    attempt: DataTypes.INTEGER,
     status: DataTypes.STRING,
-    smsProvider: DataTypes.STRING,
+    errors: DataTypes.STRING,
   }) as AttestationStatic
 
-  model.prototype.canSendSms = function() {
-    return [
-      AttestationStatus.DISPATCHING,
-      AttestationStatus.FAILED,
-      AttestationStatus.UNABLE_TO_SERVE,
-    ].includes(this.status)
+  model.prototype.key = function(): AttestationKey {
+    return { account: this.account, identifier: this.identifier, issuer: this.issuer }
+  }
+
+  model.prototype.provider = function(): string | null {
+    const pl = this.providers.split(',')
+    return this.providers ? pl[this.attempt % pl.length] : null
+  }
+
+  model.prototype.recordError = function(error: string) {
```

Comment:
where does this get used? i don't see any references to it

Reply:
`src/sms/index.ts` a couple of places

Reply:
oh gotcha -- github had defaulted the toggle of that file to closed 🙄